src/connector.rs

```diff
@@ -284,10 +263,18 @@ mod tests {
             .with_no_client_auth();
     }
 
-    async fn oneshot<S: Service<Uri>>(
-        mut service: S,
+    async fn connect(
+        https_only: bool,
         https: bool,
```

Comment:
similar nit here: two `bool`'s side-by-side is easy to get wrong at the call-site.